packages/block-editor/src/components/use-editor-feature/index.js

```diff
@@ -1,23 +1,61 @@
+/**
+ * External dependencies
+ */
+import { get, has } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { useSelect } from '@wordpress/data';
 
+/**
+ * Internal dependencies
+ */
+import { useBlockEditContext } from '../block-edit';
+
 /**
  * Hook that retrieves the setting for the given editor feature.
+ * It works with nested objects using by finding the value at path.
  *
- * @param {string} featureName The name of the feature.
+ * @param {string} featurePath The path to the feature.
  *
  * @return {any} Returns the value defined for the setting.
+ *
+ * @example
+ * ```js
+ * const isEnabled = useEditorFeature( 'typography.dropCap' );
+ * ```
  */
-export default function useEditorFeature( featureName ) {
+export default function useEditorFeature( featurePath ) {
+	const { name: blockName } = useBlockEditContext();
+
 	const setting = useSelect(
 		( select ) => {
 			const { getSettings } = select( 'core/block-editor' );
 
-			return getSettings()[ featureName ];
+			const path = featurePath.split( '.' );
```

Comment:
I made a related comment if we should pass the array paths directly https://github.com/WordPress/gutenberg/pull/22291#discussion_r425313407.
But I guess an option would be to just pass the path to lodash functions this support 'a.b' format directly and we don't need to do the split. So we would support the array format and 'a.b' internally we should probably use arrays because of the performance reason.

Reply:
```js
_.get(object, 'a[0].b.c');
// => 3
 
_.get(object, ['a', '0', 'b', 'c']);
// => 3

_.get(object, ['a', '0', 'b.c']);
// => undefined
```

You can't mix two approaches, that's why `split` is used. We can `join` those strings as an alternative and pass as a string to lodash.